snuba/replacer.py

```diff
@@ -381,3 +403,64 @@ def flush_batch(self, batch: Sequence[Replacement]) -> None:
             logger.info(
                 "Optimized %s partitions on %s" % (num_dropped, clickhouse_read.host)
             )
+
+    def _message_already_processed(self, metadata: ReplacementMessageMetadata) -> bool:
+        """
+        Figure out whether or not the message was already processed.
+
+        Check if there exists a recorded offset that took too long to execute.
+        If there is, we can conclude whether or not to process the incoming
+        message by comparing the incoming message to the recorded offset.
+        """
+        # float("-inf") implies this is the first message this worker is processing since startup
+        if self.__prolonged_offset == float("-inf"):
```

Comment:
I had `None` to begin with but mypy complained about setting an int to a Nonetype. I could use any other holding int like -2, I chose `float("-inf")` since it's a common way of denoting a default low number.

Reply:
Back to this comment, sorry for the delay.
Please do not use the float. Offsets are integers.
If you need that to be nullable, make it Optional[int] instead of int.
Turning the offset into a float while it is an int in order to avoid the nullability issue introduces a hack to solve a problem that can be solved properly with nullability.